4. As a user, I want to set up a new shopping list so I can start tracking purchased items.

3campos opened the pull request:
## Description

This code creates an accessible form that allows users to create grocery lists which are added to the Firestore database. When the user creates a list, a message appears confirming whether the creation was successful or not. It also adds the path to local storage. After creating a list, the app navigates users to the list page.

## Related Issue

closes #4 

## Acceptance Criteria

UI-related tasks:

[x] The Home view displays a form that allows users to enter the name of a shopping list and then create a list with that name.
[x] The input that accepts the name of the item has a semantic label element associated with it
[x] The user can submit this form with both the mouse and the Enter key
[x] When the user submits the form, they see a message indicating that the list either was or was not created and saved to the database.

Data-related tasks:

[x] Clicking the button uses the createList function in src/api/firebase.js to create a new shopping list for the user.
[x] The shopping list path is stored in local storage using the setListPath function.
[x] Once the list has been created and saved, the user is redirected to the List view.

## Type of Changes

## Updates

### After

https://github.com/the-collab-lab/tcl-66-smart-shopping-list/assets/105468537/5b91898a-b820-4262-b445-95b5507620bb

## Testing Steps / QA Criteria

<!-- Provide steps the other cohort members and mentors need to follow to properly test your additions. -->

1. From the code, type "npm start" in your terminal
2. Click the Sign In button and sign in using the user you have connected to firebase
3. Type a grocery list name
4. Click "Create List" or press "Enter" on your keyboard
5. Open your DevTools Window 
6. Navigate to "Application" tab and open the drop down menu for "Local Storage"
7. Check that the path to the list is added into local storage for your localhost page

StefieCaff committed:
adding new list functionality to home

StefieCaff committed:
implement setListPath function and navigation functionality -ready for testing

hun-ah reviewed (commented):
Nice work! The code is working well for me, the only thing I am wondering about is after I successfully create a list, I am redirected to the list page but I'm not seeing the message indicating I have successfully created the list. I assume this is because it is instantly navigating to the list page. You may want to use a setTimeout function to display the message first and then navigate to the list page.

3campos committed:
add 2 second delay after list creation

StefieCaff committed:
create AddList.jsx Update handleSubmit

create new component for AddList form.
swap AddList component for form in Home.jsx
update error message in catch block
clearlist name after neList is created

StefieCaff committed:
replace && operator

use ternary operator instead of && as per https://dev.to/maafaishal/avoid-operator-for-conditional-rendering-in-react-2de)

eonflower reviewed (approved):
Looks great! I like that you made the AddList into a component. Keeps the home view nice and clean :)

hun-ah reviewed (approved):
Looks good!

deeheber commented on src/components/AddList.jsx:
```diff
@@ -0,0 +1,49 @@
+	const handleSubmit = (event) => {
+		event.preventDefault();
+		try {
+			const newList = createList(user.uid, user.email, listName);
```
Not blocking: you could check if a user is present (aka if they're logged in or not) and throw up an alert prompting the user to login to create a new list if they aren't logged in in this handleSubmit function.

StefieCaff replied:
Good idea :) I have a question though, what does it mean not blocking?

Also, we had casually discussed adding functionality later--- like private vs public routes or validation of some kind. If we do that, would adding this to the handleSubmit function be duplicitous?

deeheber replied:
Update: we chatted about this via DM. Feel free to ignore that comment for now and merge when ready, I missed the convo about adding this later.

deeheber reviewed (approved):
This looks good to merge to me.

Had the one non-blocking suggestion since it wasn't part of the requirements, it's totally up to you if you want to add that in or not before merging. 🚀
